apps/server/src/controllers/integrationController.ts

```diff
@@ -222,6 +237,74 @@ export function dispatchFromAdapter(type: string, payload: unknown, source?: 'os
       return { topic: 'timer', payload: timer };
     }
 
+    // ontime/change/{eventID}/{propertyName}
+    case 'change': {
+      try {
+        if (params.length < 2) {
+          throw new Error('Invalid parameters');
+        }
+
+        const eventID = params[0];
+        const propertyName = params[1] as keyof OntimeEvent;
+
+        const isKeyOfOntimeEvent = (key: string): key is keyof OntimeEvent => {
+          return key in event;
+        };
+
+        if (!isKeyOfOntimeEvent(propertyName)) {
+          throw new Error(`Cannot update unknown event property ${propertyName}`);
+        }
+
+        const acceptedPropertyPayloadMap = {
+          title: 'string',
+          subtitle: 'string',
+          presenter: 'string',
+          note: 'string',
+          cue: 'string',
+
+          duration: 'number',
+
+          isPublic: 'boolean',
+          skip: 'boolean',
+
+          colour: 'string',
+          user0: 'string',
+          user1: 'string',
+          user2: 'string',
+          user3: 'string',
+          user4: 'string',
+          user5: 'string',
+          user6: 'string',
+          user7: 'string',
+          user8: 'string',
+          user9: 'string',
+        };
+
+        const parsePayload = (payload: unknown, propertyName: keyof OntimeEvent): boolean | string | number => {
+          switch (acceptedPropertyPayloadMap[propertyName]) {
+            case 'string': {
+              return String(payload);
+            }
+            case 'number': {
+              return Number(payload);
+            }
+            case 'boolean': {
+              return Boolean(payload);
+            }
+            default: {
+              throw new Error('Unhandled payload type');
+            }
+          }
+        };
+
+        const parsedPayload = parsePayload(payload, propertyName);
+
+        return PlaybackService.updateEvent(eventID, propertyName, parsedPayload);
```

Comment:
it is not within scope of the `playbackService` to change events

We have a function `editEvent` in `RundownService`, this function makes sure the change is made in all necessary places.
We would need to construct a patch object to pass to `editEvent`

This means that, in the case of duration, we need the logic to make make the necessary time changes to happen before we call `editEvent`. `editEvent` should not contain business logic

maybe one more function for the extracted utilities :D

Reply:
what I previously had is:

`integrationController` calls `PlaybackService` `updateEvent` method, which called `updateEvent` in `RundownService`, that called `editEvent` inside the own module, which to be fair is kinda confusing

what I can do is

have `updateEvent` to be inside another module (config?) and call things in this order:

`integrationController` -> `updateEvent` -> `editEvent`.

`updateEvent` would deal with the business logic for changing things on the update

Reply:
I agree, the playback service shouldnt have a updateEvent method since it is not in its scope to do so
The management of events is in scope of the RundownService

Reply:
I had moved`updateEvent` to `integrationController.config.ts`, should I move it to `RundownService`? problem is it will have both `editEvent` and `updateEvent` on it